src/lib.rs

```diff
@@ -139,21 +119,28 @@ impl Client {
     pub async fn get_many<I, K>(&mut self, keys: I) -> Result<Vec<Value>, Error>
     where
         I: IntoIterator<Item = K>,
-        K: AsRef<[u8]>,
+        K: AsRef<[u8]> + Clone,
     {
-        self.conn.write_all(b"get ").await?;
-        for key in keys.into_iter() {
-            self.conn.write_all(key.as_ref()).await?;
-            self.conn.write_all(b" ").await?;
-        }
-        self.conn.write_all(b"\r\n").await?;
-        self.conn.flush().await?;
+        // Groups keys by the server they belong to and then fetch from each server asyncronously
+        let server_map = keys
+            .into_iter()
+            .into_group_map_by(|key| self.ring.server_index_for(key));
+        let result = stream::iter(self.ring.servers.iter_mut())
+            .enumerate()
+            .then(move |(node_index, node)| {
+                let cloned_server_map = server_map.clone();
+
+                async move {
+                    let empty: Vec<K> = Vec::new();
+                    let keys = cloned_server_map.get(&node_index).unwrap_or(&empty);
+                    node.get_many(keys).await
```

Comment:
Can we either skip this when `keys` is empty, or add something to `Node::get_many` to return early when `keys` is empty?

Reply:
Opted to return early with an empty vector in `Node::get_many` in c9ce02c.